// nick_common_static/common/models/dashboard/bas_apply_list_model.ts
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */

import cd = require('common/models/common/base_data');
import bm = require('common/models/dashboard/bas_model');

export module Dashboard {
    /*
     * 管理同好会的接口参数
     * */
    export class BasApplyListModel extends cd.Common.BaseData {
        public applyId:number;
        public status:number;
        public intro:string;
        public message:string;
        public isSendNotice:boolean;
        public basData:bm.Dashboard.BasModel;

        public static _convert(data:any){
            var model = new Dashboard.BasApplyListModel({
                basData: bm.Dashboard.BasModel._convert(data),
                applyId: data.id,
                status: data.status,
                intro: data.intro,
                message: data.reply_message,
            });

            model.basData.copyData({
                title: data.ba_title,
                intro: data.ba_intro
            });

            return model;
        }

        constructor(data?:any) {
            super(data);
        }
    }
}
