// nick_common_static/common/models/dashboard/bas_recommend_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'common/models/common/base_data', 'common/models/dashboard/bas_model'], function (require, exports, cd, bm) {
    var Dashboard;
    (function (Dashboard) {
        /*
         * 同好会申请模型
         * */
        var BasRecommendModel = (function (_super) {
            __extends(BasRecommendModel, _super);
            function BasRecommendModel(data) {
                _super.call(this, data);
            }
            BasRecommendModel._convert = function (data) {
                var model = new Dashboard.BasRecommendModel({
                    recommendId: data.id,
                    status: data.status,
                    createAt: data.created_at,
                    updateAt: data.updated_at,
                    basData: bm.Dashboard.BasModel._convert(data)
                });
                model.basData.baId = data.ba_id;
                return model;
            };
            return BasRecommendModel;
        })(cd.Common.BaseData);
        Dashboard.BasRecommendModel = BasRecommendModel;
    })(Dashboard = exports.Dashboard || (exports.Dashboard = {}));
});
