// nick_common_static/common/models/dashboard/bas_apply_list_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'common/models/common/base_data', 'common/models/dashboard/bas_model'], function (require, exports, cd, bm) {
    var Dashboard;
    (function (Dashboard) {
        /*
         * 管理同好会的接口参数
         * */
        var BasApplyListModel = (function (_super) {
            __extends(BasApplyListModel, _super);
            function BasApplyListModel(data) {
                _super.call(this, data);
            }
            BasApplyListModel._convert = function (data) {
                var model = new Dashboard.BasApplyListModel({
                    basData: bm.Dashboard.BasModel._convert(data),
                    applyId: data.id,
                    status: data.status,
                    intro: data.intro,
                    message: data.reply_message
                });
                model.basData.copyData({
                    title: data.ba_title,
                    intro: data.ba_intro
                });
                return model;
            };
            return BasApplyListModel;
        })(cd.Common.BaseData);
        Dashboard.BasApplyListModel = BasApplyListModel;
    })(Dashboard = exports.Dashboard || (exports.Dashboard = {}));
});
