// nick_common_static/common/models/dashboard/forum_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'common/models/common/base_data', 'common/models/passport/user_model'], function (require, exports, cd, um) {
    var Dashboard;
    (function (Dashboard) {
        /*
         * 论坛模型
         * */
        var ForumModel = (function (_super) {
            __extends(ForumModel, _super);
            function ForumModel(data) {
                _super.call(this, data);
            }
            ForumModel._schema = {
                type: 'object',
                description: '热门板块管理',
                required: [
                    'title',
                    'intro',
                    'pictures',
                    'url'
                ],
                properties: {
                    forumId: {
                        type: 'number',
                        title: 'ID'
                    },
                    title: {
                        type: 'string',
                        title: '标题'
                    },
                    intro: {
                        type: 'string',
                        title: '简介',
                        maxLength: 150
                    },
                    url: {
                        type: 'string',
                        title: '链接地址'
                    },
                    pictures: {
                        type: 'array',
                        title: '热门板块图片',
                        items: {
                            type: 'object',
                            required: ['url'],
                            properties: {
                                url: {
                                    type: 'string',
                                    title: '图片地址'
                                }
                            }
                        }
                    },
                    creator: um.Passport.UserModel._schema,
                    editor: um.Passport.UserModel._schema
                }
            };
            ForumModel._convert = function (data) {
                return new Dashboard.ForumModel({
                    forumId: data.id,
                    title: data.title,
                    intro: data.intro,
                    pictures: data.pic_urls ? data.pic_urls.split(',').map(function (pic) {
                        return { url: pic };
                    }) : [],
                    url: data.url,
                    status: data.status,
                    creator: new um.Passport.UserModel({
                        user_token: data.user_token,
                        nickname: data.nickname,
                        avatar_url: data.avatar_url
                    }),
                    editor: new um.Passport.UserModel({
                        user_token: data.editor_user_token,
                        nickname: data.editor_nickname
                    }),
                    updateAt: data.updated_at,
                    createAt: data.created_at
                });
            };
            return ForumModel;
        })(cd.Common.BaseData);
        Dashboard.ForumModel = ForumModel;
    })(Dashboard = exports.Dashboard || (exports.Dashboard = {}));
});
