// nick_common_static/common/models/dashboard/notice_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'common/models/common/base_data', 'common/models/passport/user_model'], function (require, exports, cd, um) {
    var Dashboard;
    (function (Dashboard) {
        /*
         * 公告模型
         * */
        var NoticeModel = (function (_super) {
            __extends(NoticeModel, _super);
            function NoticeModel(data) {
                _super.call(this, data);
            }
            NoticeModel._convert = function (data) {
                return new Dashboard.NoticeModel({
                    noticeId: data.id,
                    content: data.content,
                    creator: new um.Passport.UserModel({
                        nickname: data.nickname,
                        user_token: data.user_token
                    }),
                    editor: new um.Passport.UserModel({
                        nickname: data.editor_nickname,
                        user_token: data.editor_user_token
                    }),
                    createAt: data.created_at
                });
            };
            NoticeModel._schema = {
                type: 'object',
                description: '公告管理',
                required: [
                    'content',
                    'url'
                ],
                properties: {
                    noticeId: {
                        type: 'number',
                        title: 'ID'
                    },
                    content: {
                        type: 'string',
                        title: '公告内容',
                        maxLength: 150
                    },
                    url: {
                        type: 'string',
                        title: '链接地址'
                    },
                    theme: {
                        type: 'string',
                        title: '主题色'
                    },
                    tag: {
                        type: 'string',
                        title: '标签'
                    },
                    creator: um.Passport.UserModel._schema,
                    editor: um.Passport.UserModel._schema
                }
            };
            return NoticeModel;
        })(cd.Common.BaseData);
        Dashboard.NoticeModel = NoticeModel;
    })(Dashboard = exports.Dashboard || (exports.Dashboard = {}));
});
