// nick_common_static/common/models/dashboard/report_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'common/models/common/base_data', 'common/models/passport/user_model', 'common/models/dashboard/post_model'], function (require, exports, cd, um, post) {
    var Dashboard;
    (function (Dashboard) {
        /*
         * 管理同好会的接口参数
         * */
        var ReportModel = (function (_super) {
            __extends(ReportModel, _super);
            function ReportModel(data) {
                _super.call(this, data);
            }
            ReportModel._convert = function (data) {
                return new Dashboard.ReportModel({
                    reportId: data.id,
                    reportType: data.report_type,
                    reportContent: data.report_content,
                    nickname: data.nickname,
                    status: data.status,
                    createAt: data.created_at,
                    updateAt: data.updated_at,
                    postData: new post.Dashboard.PostModel({
                        postId: data.post_id,
                        title: data.post_title,
                        content: data.post_content,
                        pictures: data.post_pic_urls ? data.post_pic_urls.split(',').map(function (pic) {
                            return { url: pic };
                        }) : []
                    }),
                    replyData: new post.Dashboard.PostModel({
                        postId: data.reply_id,
                        //title: data.post_title,
                        content: data.reply_content,
                        pictures: data.reply_pic_urls ? data.reply_pic_urls.split(',').map(function (pic) {
                            return { url: pic };
                        }) : []
                    }),
                    admin: new um.Passport.UserModel({
                        userToken: data.admin_user_token,
                        nickname: data.admin_nickname
                    })
                });
            };
            ReportModel._schema = {
                type: 'object',
                description: '举报',
                required: [
                    'reportType'
                ],
                properties: {
                    reportType: {
                        type: 'number',
                        title: '举报类型'
                    },
                    postType: {
                        type: 'number',
                        title: '帖子类型'
                    },
                    reportContent: {
                        type: 'string',
                        maxLength: 150,
                        title: '举报内容'
                    },
                    replyData: post.Dashboard.PostModel._schema,
                    postData: post.Dashboard.PostModel._schema
                }
            };
            return ReportModel;
        })(cd.Common.BaseData);
        Dashboard.ReportModel = ReportModel;
    })(Dashboard = exports.Dashboard || (exports.Dashboard = {}));
});
