// nick_common_static/common/models/dashboard/report_model.ts
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */

import cd = require('common/models/common/base_data');
import um = require('common/models/passport/user_model');
import post = require('common/models/dashboard/post_model');
import bas = require('common/models/dashboard/bas_model');

export module Dashboard {
    /*
     * 管理同好会的接口参数
     * */
    export class ReportModel extends cd.Common.BaseData {
        public reportId:number;
        public reportType:string;
        public postType:number;
        public reportContent:string;
        public nickname:string;
        // 0 :未处理 1:已下架 2: 已删除
        public status:boolean;
        public replyData:post.Dashboard.PostModel;
        public postData:post.Dashboard.PostModel;
        public BasData:bas.Dashboard.BasModel;
        public admin:um.Passport.UserModel;

        public static _convert(data:any):Dashboard.ReportModel{
            return new Dashboard.ReportModel({
                reportId: data.id,
                reportType: data.report_type,
                reportContent: data.report_content,
                nickname: data.nickname,
                status: data.status,
                createAt: data.created_at,
                updateAt: data.updated_at,
                postData: new post.Dashboard.PostModel({
                    postId: data.post_id,
                    title: data.post_title,
                    content: data.post_content,
                    pictures: data.post_pic_urls ? data.post_pic_urls.split(',').map((pic)=> {
                        return {url: pic};
                    }) : []
                }),
                replyData: new post.Dashboard.PostModel({
                    postId: data.reply_id,
                    //title: data.post_title,
                    content: data.reply_content,
                    pictures: data.reply_pic_urls ? data.reply_pic_urls.split(',').map((pic)=> {
                        return {url: pic};
                    }) : []
                }),
                admin:new um.Passport.UserModel({
                    userToken: data.admin_user_token,
                    nickname: data.admin_nickname
                })
            });
        }

        public static _schema:any = {
            type: 'object',
            description: '举报',
            required: [
                'reportType'
            ],
            properties: {
                reportType: {
                    type: 'number',
                    title: '举报类型'
                },
                postType: {
                    type: 'number',
                    title: '帖子类型'
                },
                reportContent: {
                    type: 'string',
                    maxLength: 150,
                    title: '举报内容'
                },
                replyData: post.Dashboard.PostModel._schema,
                postData: post.Dashboard.PostModel._schema
            }
        };

        constructor(data?:any) {
            super(data);
        }
    }
}
