// nick_common_static/common/models/passport/register_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'common/models/common/base_data'], function (require, exports, bd) {
    var Passport;
    (function (Passport) {
        /*
         * 注册模型
         * */
        var RegisterModel = (function (_super) {
            __extends(RegisterModel, _super);
            function RegisterModel(data) {
                _super.call(this, data);
            }
            RegisterModel._schema = {
                type: "object",
                properties: {
                    username: {
                        type: 'string',
                        minLength: 6,
                        maxLength: 30,
                        required: true,
                        title: '用户名'
                    },
                    password: {
                        type: 'string',
                        title: '密码',
                        minLength: 6,
                        maxLength: 30
                    },
                    passwordAgain: {
                        type: 'string',
                        title: '重复密码',
                        minLength: 6,
                        maxLength: 30
                    }
                }
            };
            return RegisterModel;
        })(bd.Common.BaseData);
        Passport.RegisterModel = RegisterModel;
    })(Passport = exports.Passport || (exports.Passport = {}));
});
