// nick_common_static/common/models/passport/register_model.ts
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */

import bd = require('common/models/common/base_data');

export module Passport {
    /*
     * 注册模型
     * */
    export class RegisterModel extends bd.Common.BaseData {
        public username:string;
        public password:string;
        public passwordAgain:string;
        public smsToken:string;

        public static _schema= {
            type: "object",
            properties: {
                username: {
                    type: 'string',
                    minLength: 6,
                    maxLength: 30,
                    required: true,
                    title: '用户名',
                },
                password: {
                    type: 'string',
                    title: '密码',
                    minLength: 6,
                    maxLength: 30,
                },
                passwordAgain: {
                    type: 'string',
                    title: '重复密码',
                    minLength: 6,
                    maxLength: 30,
                }
            }
        };

        constructor(data?:any) {
            super(data);
        }
    }
}
