// nick_common_static/common/config/config.js
/**
 * Created by NICK on 15/10/28.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
define(["require", "exports"], function (require, exports) {
    var ProductConfig = (function () {
        function ProductConfig() {
            this.passport_api_url = 'http://pt.p7game.com/api';
            this.dashboard_api_url = 'http://pt.p7game.com/api';
            this.putong_api_url = 'http://pt.p7game.com/api';
            this.facemash_api = 'http://pt.p7game.com/api';
            this.shiji_api = 'http://pt.p7game.com/api';
            this.prefix = 'pro_';
            this.expire_in = 86400;
            this.app_token = 'iDnRcF7-';
        }
        return ProductConfig;
    })();
    var DevelopConfig = (function () {
        function DevelopConfig() {
            this.passport_api_url = 'http://staging.pt.p7game.com/api';
            this.dashboard_api_url = 'http://staging.pt.p7game.com/api';
            this.putong_api_url = 'http://staging.pt.p7game.com/api';
            this.facemash_api = 'http://staging.pt.p7game.com/api';
            this.shiji_api = 'http://staging.pt.p7game.com/api';
            this.permiss_api = 'http://localhost:1337';
            this.prefix = 'dev_';
            this.app_token = 'iDnRcF7-';
            this.expire_in = 86400;
        }
        return DevelopConfig;
    })();
    exports.config = {
        pro: new ProductConfig(),
        dev: new DevelopConfig()
    };
});
