// nick_common_static/common/models/moemash/picture_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'common/models/common/base_data', 'common/models/passport/user_model'], function (require, exports, bd, user) {
    var Moemash;
    (function (Moemash) {
        var PictureModel = (function (_super) {
            __extends(PictureModel, _super);
            function PictureModel(data) {
                _super.call(this, data);
            }
            PictureModel._convert = function (data) {
                return new Moemash.PictureModel({
                    picId: data.id,
                    imageKey: data.image_key,
                    pkCount: data.pk,
                    score: data.score,
                    isCrawler: data.is_crawler,
                    creator: new user.Passport.UserModel({
                        userToken: data.user_token,
                        nickname: data.nickname
                    })
                });
            };
            PictureModel._schema = {
                type: 'object',
                description: '萌战管理',
                required: [
                    'score',
                    'pkCount',
                ],
                properties: {
                    picId: {
                        type: 'number',
                        title: 'ID'
                    },
                    score: {
                        type: 'number',
                        title: '分数'
                    },
                    pkCount: {
                        type: 'number',
                        title: 'PK次数'
                    }
                }
            };
            return PictureModel;
        })(bd.Common.BaseData);
        Moemash.PictureModel = PictureModel;
    })(Moemash = exports.Moemash || (exports.Moemash = {}));
});
