// nick_common_static/common/directives/gridmenu_directive.ts
/**
 * Created by NICK on 15/6/19.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */


import ref= require('ref');
import {Directives} from "./directives";

export class GridMenuDirective {
    public static _name:string = "gridmenu";

    public static directive:Array<any> = [()=> {
        var directive:ng.IDirective = {};

        directive.require = "^?ngModel";
        directive.scope = {
            tools: '=tools',
            isBusy: '=isBusy',
            openMenu: '&openMenu',
            item: '=item',
            width: '@width'
        };
        directive.replace = false;
        directive.template = "<md-menu>\n" +
            "    <md-button class=\"md-icon-button\"\n" +
            "               ng-disabled=\"isBusy\"\n" +
            "               ng-click=\"openMenu()($mdOpenMenu,$event)\">\n" +
            "        <ng-md-icon icon=\"{{ isBusy?'not_interested':'keyboard_arrow_down' }}\"></ng-md-icon>\n" +
            "    </md-button>\n" +
            "    <md-menu-content width=\"{{ width }}\">\n" +
            "        <md-menu-item ng-repeat=\"tool in tools\">\n" +
            "            <md-button ng-click=\"tool.onClick(item,$event)\">\n" +
            "                <ng-md-icon icon=\"{{ tool.icon }}\"></ng-md-icon>\n" +
            "                {{ tool.title }}\n" +
            "            </md-button>\n" +
            "        </md-menu-item>\n" +
            "    </md-menu-content>\n" +
            "</md-menu>";
        directive.link = ($scope)=> {
            !$scope['width'] && ($scope['width'] = 6);
        };

        return directive;
    }];
}
