// nick_common_static/common/directives/lightbox_directive.js
/**
 * Created by NICK on 15/10/13.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'common/common/controller/base_material_controller', 'common/common/controller/popup_material_controller'], function (require, exports, common, popup) {
    var LightboxProvider = (function (_super) {
        __extends(LightboxProvider, _super);
        function LightboxProvider() {
            _super.apply(this, arguments);
        }
        LightboxProvider._name = "lightbox";
        LightboxProvider.provider = [function () {
                var provider = {
                    $get: ['$rootScope', '$mdDialog',
                        function ($rootScope, $mdDialog) {
                            function showImage(index, images, $event) {
                                var dialogOptions = {};
                                dialogOptions.controller = 'LightboxController';
                                dialogOptions.controllerAs = 'lightboxCtl';
                                dialogOptions.template = "<md-dialog flex-auto style=\"height:100%;width: 100%;\" layout-padding aria-label=\"\">\n" +
                                    "    <md-dialog-content flex layout=\"row\" layout-padding ng-style=\"$root.getBackground(lightboxCtl.images[lightboxCtl.index].imageKey)\" style=\"background-size:contain; background-position:center center; background-repeat: no-repeat;\">\n" +
                                    //"        <md-content flex-auto layout=\"row\" ng-style=\"$root.getBackground(lightboxCtl.images[lightboxCtl.index].imageKey)\" style=\"background-size:contain;background-repeat: no-repeat;\" layout-align=\"center center\">\n" +
                                    //"            <img style=\"width: 100%;visibility: hidden;\" ng-src=\"{{lightboxCtl.images[lightboxCtl.index].imageKey}}\">\n" +
                                    //"        </md-content>\n" +
                                    "    </md-dialog-content>\n" +
                                    "\n" +
                                    "    <div class=\"md-actions\" layout=\"row\">\n" +
                                    "        <md-button class=\"md-icon-button\" ng-click=\"lightboxCtl.prev()\">\n" +
                                    "            <ng-md-icon icon=\"navigate_before\"></ng-md-icon>\n" +
                                    "        </md-button>\n" +
                                    "        <span flex></span>\n" +
                                    "        <md-button class=\"md-icon-button\" ng-click=\"lightboxCtl.next()\">\n" +
                                    "            <ng-md-icon icon=\"navigate_next\"></ng-md-icon>\n" +
                                    "        </md-button>\n" +
                                    "    </div>\n" +
                                    "</md-dialog>";
                                dialogOptions.targetEvent = $event;
                                dialogOptions.clickOutsideToClose = true;
                                dialogOptions.resolve = {
                                    'images': function () {
                                        return images;
                                    },
                                    'index': function () {
                                        return index || 0;
                                    }
                                };
                                $mdDialog.show(dialogOptions);
                            }
                            return {
                                showImage: showImage
                            };
                        }]
                };
                return provider;
            }];
        return LightboxProvider;
    })(common.BaseController);
    exports.LightboxProvider = LightboxProvider;
    var LightboxController = (function (_super) {
        __extends(LightboxController, _super);
        function LightboxController() {
            _super.call(this, arguments);
        }
        LightboxController.prototype.prev = function () {
            if (this.index == 0) {
                this.index = this.images.length - 1;
                return;
            }
            this.index--;
        };
        LightboxController.prototype.next = function () {
            if (this.index == this.images.length - 1) {
                this.index = 0;
                return;
            }
            this.index++;
        };
        LightboxController.$inject = ['$rootScope', '$scope', '$q', '$mdToast', '$mdDialog', 'images', 'index'];
        LightboxController._name = 'LightboxController';
        return LightboxController;
    })(popup.PopupController);
    exports.LightboxController = LightboxController;
    var LightboxDirective = (function (_super) {
        __extends(LightboxDirective, _super);
        function LightboxDirective() {
            _super.apply(this, arguments);
        }
        LightboxDirective._name = "lightbox";
        LightboxDirective.directive = ['$rootScope', 'lightbox', function ($rootScope, lightbox) {
                var directive = {
                    require: '^?ngModel',
                    scope: true,
                    replace: true
                };
                directive.link = function ($scope, $element, $attrs, $ctrl) {
                    $rootScope.$on('showImages', function (event, data) {
                        lightbox.showImage(data.index, data.images, data.$event);
                    });
                };
                return directive;
            }];
        return LightboxDirective;
    })(common.BaseController);
    exports.LightboxDirective = LightboxDirective;
});
