// nick_common_static/common/services/dashboard_service.ts
/**
 * Created by NICK on 15/6/17.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */

import ref = require('ref');

export class DashboardService {
    public static _name:string = 'DashboardService';

    public static factory:Array<any> = ['$rootScope', '$q', '$http', 'config', ($rootScope, $q, $http, config):any=> {
        return {
            /*
             * 管理同好会
             * */
            list_bas: function (basListQuery:Dashboard.BasListQuery) {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/list_bas',
                    params: {
                        page: basListQuery.page,
                        per_page: basListQuery.pageCount,
                        filter_ba_id: basListQuery.baId,
                        filter_ba_title: basListQuery.title
                    },
                    needToken: true,
                    needCancel: true
                }).success(function (data) {

                });

                return promise;
            },
            /*
             *  同好会申请列表
             * */
            list_ba_applies: function (basListQuery:Dashboard.BasListQuery) {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/list_ba_applies',
                    params: {
                        page: basListQuery.page,
                        per_page: basListQuery.pageCount
                    },
                    needToken: true,
                    needCancel: true
                }).success(function (data) {

                });

                return promise;
            },
            /*
             *  审核一个同好会申请
             * */
            check_a_ba_apply: function (data:Dashboard.BasApplyModel) {
                var promise = $http({
                        method: 'POST',
                        url: config.dashboard_api_url + '/putong_admin/check_a_ba_apply',
                        data: {
                            ba_apply_id: data.applyId,
                            admin_nickname: data.nickname,
                            to_status: data.status,
                            reply_message: data.message,
                            is_send_notice: data.isSendNotice
                        },
                        needToken: true
                    }
                );

                return promise;
            },
            /*
             * 创建一个同好会
             * */
            add_a_ba: function (basData:Dashboard.BasModel) {
                var promise = $http({
                        method: 'POST',
                        url: config.dashboard_api_url + '/putong_admin/add_a_ba',
                        data: {
                            bazhu_nickname: basData.creator.nickname,
                            title: basData.title,
                            intro: basData.intro,
                            ba_avatar_url: basData.avatarUrl,
                            background_url: basData.backgroundUrl,
                            theme: basData.theme
                        },
                        needToken: true
                    }
                );

                return promise;
            },
            /*
             *  下架同好会
             * */
            off_bas: function (ids:number[]) {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/off_bas',
                    data: {
                        ba_ids_str: ids.join(',')
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 上架同好会
             * */
            pub_bas: function (ids:number[]) {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/pub_bas',
                    data: {
                        ba_ids_str: ids.join(',')
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 显示同好会详情
             * params@id : 同好会id
             * */
            show_a_ba: function (basData:Dashboard.BasModel) {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/show_a_ba',
                    params: {
                        ba_id: basData.baId
                    },
                    needToken: true,
                }).success(function (data) {
                    // data = new Dashboard.BasData();
                });

                return promise;
            },
            /*
             * 更新一个同好会
             * */
            update_a_ba: function (basData:Dashboard.BasModel) {
                var promise = $http({
                        method: 'POST',
                        url: config.dashboard_api_url + '/putong_admin/update_a_ba',
                        data: {
                            ba_id: basData.baId,
                            title: basData.title,
                            intro: basData.intro,
                            ba_avatar_url: basData.avatarUrl,
                            background_url: basData.backgroundUrl,
                            theme: basData.theme
                        },
                        needToken: true
                    }
                );

                return promise;
            },
            /*
             * 添加一个吧主
             * */
            add_a_bazhu: function (id, nickname) {
                var promise = $http({
                        method: 'POST',
                        url: config.dashboard_api_url + '/putong_admin/add_a_bazhu',
                        data: {
                            ba_id: id,
                            bazhu_nickname: nickname
                        },
                        needToken: true
                    }
                );

                return promise;
            },
            /*
             *  删除一个吧主
             * */
            del_a_bazhu: function (id, nickname) {
                var promise = $http({
                        method: 'POST',
                        url: config.dashboard_api_url + '/putong_admin/del_a_bazhu',
                        data: {
                            ba_id: id,
                            bazhu_nickname: nickname
                        },
                        needToken: true
                    }
                );

                return promise;
            },
            /*
             * 管理帖子
             * */
            list_posts: function (query:Dashboard.PostListQuery) {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/list_posts',
                    params: {
                        page: query.page,
                        per_page: query.pageCount,
                        filter_ba_id: query.baId,
                        filter_post_id: query.postId,
                        filter_user_token: query.userToken
                    },
                    needToken: true,
                    needCancel: true
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 下架帖子
             * */
            off_a_post: function (ids:Array<number>) {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/off_posts',
                    data: {
                        post_ids_str: ids.join(',')
                    },
                    needToken: true
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 上架帖子
             * */
            pub_posts: function (ids:Array<number>) {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/pub_posts',
                    data: {
                        post_ids_str: ids.join(',')
                    },
                    needToken: true
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 显示帖子详情
             * params@id : 同好会id
             * */
            show_a_post: function (data:Dashboard.PostModel) {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/show_a_post',
                    params: {
                        post_id: data.postId
                    },
                    needToken: true,
                }).success(function (data) {
                    // data = new Dashboard.BasData();
                });

                return promise;
            },
            /*
             * 添加一个链接
             * */
            add_a_post_link: (data:Dashboard.PostLinkModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/add_a_post_link',
                    data: {
                        post_id: data.postId,
                        content_type: data.contentType,
                        title: data.title,
                        url: data.url,
                        to_post_id: data.toPostId,
                        to_ba_id: data.toBaId,
                        to_user_token: data.toUserToken
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 删除一个链接
             * */
            del_a_post_link: (data:Dashboard.PostLinkModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/del_a_post_link',
                    data: {
                        post_link_id: data.linkId
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 更新了一个链接
             * */
            update_a_post_link: (data:Dashboard.PostLinkModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/update_a_post_link',
                    data: {
                        post_link_id: data.linkId,
                        //post_id: data.postId,
                        content_type: data.contentType,
                        title: data.title,
                        url: data.url,
                        to_post_id: data.toPostId,
                        to_ba_id: data.toBaId,
                        to_user_token: data.toUserToken
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 发帖
             * */
            add_a_post: (data:Dashboard.PostModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/add_a_post',
                    data: {
                        louzhu_nickname: data.creator.nickname,
                        ba_id: data.basData.baId,
                        title: data.title,
                        content: data.content,
                        location: data.location,
                        platform: data.platform,
                        pic_urls: data.pictures.map((pic)=> {
                            return pic['url'];
                        }).join(',')
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 更新一个帖子
             * */
            update_a_post: (data:Dashboard.PostModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/update_a_post',
                    data: {
                        post_id: data.postId,
                        title: data.title,
                        content: data.content,
                        location: data.location,
                        pic_urls: data.pictures.map((pic)=> {
                            return pic['url'];
                        }).join(',')
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 举报列表
             * */
            list_report_posts: (query:Common.QueryBase)=> {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/list_report_posts',
                    params: {
                        page: query.page,
                        per_page: query.pageCount,
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 处理一个举报
             * */
            check_a_report_post: (data:Dashboard.ReportDealModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/check_a_report_post',
                    data: {
                        report_post_id: data.reportData.reportId,
                        admin_nickname: data.nickname,
                        to_status: data.status
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 跳转列表
             * */
            list_all_jumps: ()=> {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/list_all_jumps',
                    needToken: true,
                }).success(function () {

                });

                return promise;
            },
            /*
             * 设置一个跳转
             * */
            set_a_jumps: (data:Dashboard.JumpModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/set_a_jumps',
                    data: {
                        key: data.key,
                        label: data.label,
                        ios_weixin_url: data.weixinIosUrl,
                        android_weixin_url: data.weixinAndroidUrl,
                        apple_store_url: data.appStoreUrl,
                        apk_url: data.apkUrl,
                        default_url: data.defaultUrl
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 删除一个跳转
             * */
            del_a_jumps: (data:Dashboard.JumpModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/del_a_jumps',
                    data: {
                        key: data.key,
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 列表 首页顶图
             * */
            list_index_pictures: (data:Common.QueryBase)=> {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/list_index_pictures',
                    data: {
                        page: data.page,
                        per_page: data.pageCount
                    },
                    needToken: true,
                }).success(function () {

                });

                return promise;
            },
            /*
             * 新建一个图片
             * */
            add_a_index_picture: (data:Dashboard.TopPictureModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/add_a_index_picture',
                    data: {
                        content_type: data.contentType,
                        picture_url: data.imageKey,
                        title: data.title,
                        intro: data.intro,
                        url: data.url,
                        to_post_id: data.toPostId,
                        to_ba_id: data.toBaId,
                        to_user_token: data.toUserToken
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 更新图片
             * */
            update_a_index_picture: (data:Dashboard.TopPictureModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/update_a_index_picture',
                    data: {
                        index_picture_id: data.picId,
                        content_type: data.contentType,
                        picture_url: data.imageKey,
                        title: data.title,
                        intro: data.intro,
                        url: data.url,
                        to_post_id: data.toPostId,
                        to_ba_id: data.toBaId,
                        to_user_token: data.toUserToken
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 删除图片
             * */
            del_a_index_picture: (data:Dashboard.TopPictureModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/del_a_index_picture',
                    data: {
                        index_picture_id: data.picId
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 上架一个图片
             * */
            publish_a_index_picture: (data:Dashboard.TopPictureModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/publish_a_index_picture',
                    data: {
                        index_picture_id: data.picId,
                        position: data.position
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 清空一个图片位
             * */
            clear_a_index_picture_position: (data:Dashboard.TopPictureModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/clear_a_index_picture_position',
                    data: {
                        position: data.position
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 预览图片
             * */
            preview_index_pictures: ()=> {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/preview_index_pictures',
                    needToken: true,
                }).success(function () {

                });

                return promise;
            },
            /*
             * 热门板块列表
             * */
            list_bbs_hot_forums: ()=> {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/list_bbs_hot_forums',
                    needToken: true,
                }).success(function () {

                });

                return promise;
            },
            /*
             * 添加一个板块
             * */
            add_a_bbs_hot_forum: (data:Dashboard.ForumModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/add_a_bbs_hot_forum',
                    data: {
                        title: data.title,
                        intro: data.intro,
                        pic_urls: data.pictures.map((pic)=> {
                            return pic['url'];
                        }).join(','),
                        url: data.url
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 删除一个板块
             * */
            del_a_bbs_hot_forum: (data:Dashboard.ForumModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/del_a_bbs_hot_forum',
                    data: {
                        hot_forum_id: data.forumId
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 更新一个板块
             * */
            update_a_bbs_hot_forum: (data:Dashboard.ForumModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/update_a_bbs_hot_forum',
                    data: {
                        hot_forum_id: data.forumId,
                        title: data.title,
                        intro: data.intro,
                        pic_urls: data.pictures.map((pic)=> {
                            return pic['url'];
                        }).join(','),
                        url: data.url
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 下架板块
             * */
            off_bbs_hot_forums: (data:Array<number>)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/off_bbs_hot_forums',
                    data: {
                        hot_forum_ids_str: data.join(',')
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 上架板块
             * */
            on_bbs_hot_forums: (data:Array<number>)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/on_bbs_hot_forums',
                    data: {
                        hot_forum_ids_str: data.join(',')
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 热门帖子列表
             * */
            list_bbs_hot_posts: ()=> {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/list_bbs_hot_posts',
                    needToken: true,
                }).success(function () {

                });

                return promise;
            },
            /*
             * 添加一个帖子
             * */
            add_a_bbs_hot_post: (data:Dashboard.PostModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/add_a_bbs_hot_post',
                    data: {
                        title: data.title,
                        content: data.content,
                        pic_urls: data.pictures.map((pic)=> {
                            return pic['url'];
                        }).join(','),
                        url: data.url
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 删除一个帖子
             * */
            del_a_bbs_hot_post: (data:Dashboard.PostModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/del_a_bbs_hot_post',
                    data: {
                        hot_post_id: data.postId,
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 更新一个帖子
             * */
            update_a_bbs_hot_post: (data:Dashboard.PostModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/update_a_bbs_hot_post',
                    data: {
                        hot_post_id: data.postId,
                        title: data.title,
                        content: data.content,
                        pic_urls: data.pictures.map((pic)=> {
                            return pic['url'];
                        }).join(','),
                        url: data.url
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 下架帖子
             * */
            off_bbs_hot_posts: (data:Array<number>)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/off_bbs_hot_posts',
                    data: {
                        hot_post_ids_str: data.join(',')
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 下架帖子
             * */
            on_bbs_hot_posts: (data:Array<number>)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/on_bbs_hot_posts',
                    data: {
                        hot_post_ids_str: data.join(',')
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 预览论坛入口
             * */
            preview_portal_bbs: ()=> {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/preview_portal_bbs',
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 推荐同好会列表
             * */
            list_push_bas: ()=> {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/list_push_bas',
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 添加一个 发现页推荐同好会
             * */
            add_a_push_ba: (data:Dashboard.BasModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/add_a_push_ba',
                    data: {
                        ba_id: data.baId,
                        title: data.title,
                        intro: data.intro,
                        ba_avatar_url: data.avatarUrl,
                        background_url: data.backgroundUrl,
                        theme: data.theme
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 删除一个 发现页推荐同好会
             * */
            del_a_push_ba: (data:Dashboard.BasRecommendModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/del_a_push_ba',
                    data: {
                        push_ba_id: data.recommendId
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 更新一个 发现页推荐同好会
             * */
            update_a_push_ba: (data:Dashboard.BasRecommendModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/update_a_push_ba',
                    data: {
                        push_ba_id: data.recommendId,
                        title: data.basData.title,
                        intro: data.basData.intro,
                        ba_avatar_url: data.basData.avatarUrl,
                        background_url: data.basData.backgroundUrl,
                        theme: data.basData.theme
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 下架发现页推荐同好会
             * */
            off_push_bas: (data:Array<any>)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/off_push_bas',
                    data: {
                        push_ba_ids_str: data.join(',')
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 上架发现页推荐同好会
             * */
            on_push_bas: (data:Array<any>)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/on_push_bas',
                    data: {
                        push_ba_ids_str: data.join(',')
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            /*
             * 发布 发现页推荐同好会
             * */
            publish_push_bas: (data:Dashboard.BasModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/publish_push_bas',
                    data: {
                        limit: data.limit
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            preview_top_bas: ()=> {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/preview_top_bas',
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            list_public_notices: ()=> {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/list_public_notices',
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            add_a_public_notice: (data:Dashboard.NoticeModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/add_a_public_notice',
                    data: {
                        content: data.content,
                        theme: data.theme,
                        tag: data.tag,
                        url: data.url,
                        uer_token: null,
                        nickname: null
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            del_a_public_notice: (data:Dashboard.NoticeModel)=> {
                var promise = $http({
                    method: 'POST',
                    url: config.dashboard_api_url + '/putong_admin/del_a_public_notice',
                    data: {
                        public_notice_id: data.noticeId
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            },
            show_a_ba_apply: (data:Dashboard.BasApplyListModel)=> {
                var promise = $http({
                    method: 'GET',
                    url: config.dashboard_api_url + '/putong_admin/show_a_ba_apply',
                    params: {
                        ba_apply_id: data.applyId
                    },
                    needToken: true,
                }).success(function (data) {

                });

                return promise;
            }
        };
    }];
}
